### PR TITLE
Add com.yubico.yubioath

### PR DESCRIPTION
**Pull request by ai**
This PR is based on [the previous attempt](https://github.com/flathub/flathub/pull/1126) of adding [YubiOauth](https://www.yubico.com/products/services-software/download/yubico-authenticator/). I keep all @valentindavid commits in the history to give him credits.

I send multiple PRs to upstream. They finally release it and I am sending PR.

I can support the YubiOath app since I am using it every day.

### CODE REVIEW

**Comment by ai** [comment]
@LeandroStanger @barthalion I fixed the previous PR according to your comments https://github.com/flathub/flathub/pull/1126

**Comment by ai** [comment]
bot, build com.yubico.YubiOath

**Comment by ai** [comment]
bot, build com.yubico.YubiOath

**Comment by ai** [comment]
bot, build com.yubico.YubiOath

**Comment by ai** [comment]
bot, build com.yubico.YubiOath

**Comment by ai** [comment]
bot, build com.yubico.YubiOath

**Comment by ai** [comment]
bot, build com.yubico.YubiOath

**Comment by ai** [comment]
bot, stop

**Comment by bilelmoussaoui** [comment]
your app-id is com.yubico.yubioath while the manifest is still named `com.yubico.YubiOath.json`. One of the two should be updated

**Comment by ai** [comment]
@bilelmoussaoui thanks! Fixed.

Can you stop this task? https://flathub.org/builds/#/builders/32/builds/18436

**Comment by bilelmoussaoui** [comment]
bot, build com.yubico.yubioath

**Comment by ai** [comment]
@bilelmoussaoui can I ask you to help me with patch file to cut wrong `<release>` node from upstream `appschema`. I already send them [PR](https://github.com/Yubico/yubioath-desktop/pull/568) but YubiCo will release it only a few months later.

I loaded this patch file from the pull request (and I am not sure that I did it correctly).

**Comment by barthalion** [comment]
bot, build com.yubico.yubioath

**Comment by ai** [comment]
@barthalion any idea why my patch file does not fix `releases`?

I tested it locally and it removed wrong release.

```xml
  <releases>
      <release version="5.0.3" date="2020-04-14">
      </release>
      <release version="5.0.3-beta2" type="development" date="2020-03-27">
      </release>
      <release version="5.0.2" date="2020-01-29">
      <description>
        <ul>
          <li>Allow spaces in secret key when adding accounts</li>
          <li>UI: Move favorite button to the credential card</li>
          <li>UI: Improved scrolling behaviour</li>
          <li>Windows: Improve how .dll files are loaded, thanks to Marius Gabriel Mihai for reporting this!</li>
          <li>Windows: Initial support for .msi installers. If upgrading from the .exe installer, make sure to uninstall the older version before using the .msi</li>
        </ul>
      </description>
    </release>
```

Is it correct appdata file?

**Comment by bilelmoussaoui** [comment]
You can run the appdata validation test locally, see https://github.com/flathub/flathub/wiki/AppData-Guidelines#use-flathubs-appstream-util

**Comment by barthalion** [comment]
It fails to validate because 5.0.3 is considered lower than 5.0.3-beta2 by appstream-glib (and appstreamcli too). It will work when you change it to `5.0.3~beta2`. That being said, I can't find version comparison rules in the spec but it's Sunday.

**Comment by ai** [comment]
bot, build com.yubico.yubioath

**Comment by ai** [comment]
bot, build com.yubico.yubioath

**Comment by ai** [comment]
bot, build com.yubico.yubioath

**Comment by ai** [comment]
bot, build com.yubico.yubioath

**Comment by ai** [comment]
bot, build com.yubico.yubioath

**Comment by ai** [comment]
@barthalion I finally fixed diff file =^_^=

I built and tested a new version. It works with my key.

**Comment by nedrichards** [comment]
When I start it I get lots of:

> qml: failed to update readers: no_pcscd

and then when I plug in my key I get

> qml: refreshing devices failed: no_pcscd

**Comment by ai** [comment]
@nedrichards here is my output:

```
➜ flatpak run com.yubico.yubioath 
QSocketNotifier: Can only be used with threads started with QThread
Got library name:  "/app/lib/qml/io/thp/pyotherside/libpyothersideplugin.so"
Qt Quick Layouts: Detected recursive rearrange. Aborting after two iterations.
Qt Quick Layouts: Detected recursive rearrange. Aborting after two iterations.
Qt Quick Layouts: Detected recursive rearrange. Aborting after two iterations.
```

Where I can try to find `no_pcscd` messages?

**Comment by nedrichards** [comment]
I just ran flatpak run com.yubico.yubioath same as you. I'm running on Endless OS 3.8.0 beta2

```
nick@nedr-yoga:$ flatpak run com.yubico.yubioath
Gtk-Message: 19:29:56.775: Failed to load module "canberra-gtk-module"
Gtk-Message: 19:29:56.776: Failed to load module "canberra-gtk-module"
Qt: Session management error: Could not open network socket
Got library name:  "/app/lib/qml/io/thp/pyotherside/libpyothersideplugin.so"
qml: failed to update readers: no_pcscd
qml: failed to update readers: no_pcscd
qml: failed to update readers: no_pcscd
qml: failed to update readers: no_pcscd
Qt Quick Layouts: Detected recursive rearrange. Aborting after two iterations.
Qt Quick Layouts: Detected recursive rearrange. Aborting after two iterations.
```

**Comment by ai** [comment]
Any ideas why we have different behavior?

I am not so good in Qt 😓

**Comment by ai** [comment]
@nedrichards does YubiOAth works on your machine without Flatpak?

**Comment by bilelmoussaoui** [comment]
> I just ran flatpak run com.yubico.yubioath same as you. I'm running on Endless OS 3.8.0 beta2
> 
> ```
> nick@nedr-yoga:$ flatpak run com.yubico.yubioath
> Gtk-Message: 19:29:56.775: Failed to load module "canberra-gtk-module"
> Gtk-Message: 19:29:56.776: Failed to load module "canberra-gtk-module"
> Qt: Session management error: Could not open network socket
> Got library name:  "/app/lib/qml/io/thp/pyotherside/libpyothersideplugin.so"
> qml: failed to update readers: no_pcscd
> qml: failed to update readers: no_pcscd
> qml: failed to update readers: no_pcscd
> qml: failed to update readers: no_pcscd
> Qt Quick Layouts: Detected recursive rearrange. Aborting after two iterations.
> Qt Quick Layouts: Detected recursive rearrange. Aborting after two iterations.
> ```

I think that's due to an old version of Flatpak. Smart card supports (`--socket=pcscd`) was added in https://github.com/flatpak/flatpak/pull/2778

Tested and works fine with my key here. We should  set a `--require-version`  but I have no idea in which release the PCSC support landed.

**Comment by ai** [comment]
According to the [commit](https://github.com/flatpak/flatpak/commit/8ce289b4c31fe3582613cc3b07239721af725be5) and [changelog](https://github.com/flatpak/flatpak/releases/tag/1.3.2) it was `1.3.2`.

@bilelmoussaoui is it the right way to set the version limit? a3ce03e

**Comment by nedrichards** [comment]
OK yes, I have 1.6.2 but I suspect I'm missing something else in the base os.

**Comment by barthalion** [comment]
@ai It should be. Have you tested it locally?

**Comment by barthalion** [comment]
Also have you contacted upstream team if they are interested in co-maintaining their app on Flathub?

**Comment by ai** [comment]
> Also have you contacted upstream team if they are interested in co-maintaining their app on Flathub?

Unfortunately, they [do not want](https://github.com/Yubico/yubioath-desktop/issues/506) to support Faltpak package.

> We (Yubico) currently won't commit to maintaining a Flatpak, however, there is a community effort going on here: flathub/flathub#1126

**Comment by ai** [comment]
bot, build com.yubico.yubioath

**Comment by ai** [comment]
@barthalion 

> It should be. Have you tested it locally?

Everything works locally

**Comment by nedrichards** [comment]
OK, yeah, this is because my computer doesn't have the pscsd daemon.

File: com.yubico.yubioath.json [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+  "runtime": "org.kde.Platform",
+  "sdk": "org.kde.Sdk",
+  "runtime-version": "5.14",
+  "command": "yubioath-desktop",
+  "cleanup": [
+    "/include",
+    "/lib/pkgconfig"
+  ],
+  "finish-args": [
+    "--socket=x11",
```
Comment (nedrichards) [inline com.yubico.yubioath.json:12]: this needs --share=ipc to go along with the X11 socket.

Comment (ai) [reply com.yubico.yubioath.json:12]: @nedrichards like this? 2f65c56

**Comment by ai** [comment]
Build failed because of GitHub static server was down.

I was run it again a hour later, when GitHub will fix its problems.

**Comment by ai** [comment]
bot, build com.yubico.yubioath

**Comment by ai** [comment]
@bilelmoussaoui I tested fresh build, everything works

**Pull request closed without merging**